generator/internal/genclient/translator/protobuf/protobuf.go

```diff
@@ -207,17 +207,29 @@ func processMessage(state *genclient.APIState, m *descriptorpb.DescriptorProto,
 		e := processEnum(state, e, mFQN, message)
 		message.Enums = append(message.Enums, e)
 	}
-	// TODO(codyoss): https://github.com/googleapis/google-cloud-rust/issues/39
+	for _, oneof := range m.OneofDecl {
+		oneOfs := &genclient.OneOf{
+			Name:   oneof.GetName(),
+			ID:     mFQN + "." + oneof.GetName(),
+			Parent: message,
+		}
+		message.OneOfs = append(message.OneOfs, oneOfs)
+	}
 	for _, mf := range m.Field {
 		field := &genclient.Field{
-			Name:     mf.GetName(),
-			ID:       mFQN + "." + mf.GetName(),
-			JSONName: mf.GetJsonName(),
-			Optional: mf.Proto3Optional != nil && *mf.Proto3Optional,
-			Repeated: mf.Label != nil && *mf.Label == descriptorpb.FieldDescriptorProto_LABEL_REPEATED,
+			Name:            mf.GetName(),
+			ID:              mFQN + "." + mf.GetName(),
+			JSONName:        mf.GetJsonName(),
+			Optional:        mf.Proto3Optional != nil && *mf.Proto3Optional,
```

Comment:
Maybe time to refactor that expression to a variable?

Reply:
done